File: android/app/src/main/java/com/expenses/expensesdetailview/composables/BoxRow.kt
package com.expenses.expensesdetailview.composables

import androidx.compose.foundation.layout.*
import androidx.compose.material.Divider
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import com.expenses.expensesdetailview.theme.Colors
import com.expenses.expensesdetailview.theme.FontSizes
import com.expenses.expensesdetailview.theme.Sizes

@Composable
fun BoxRow(label: String, value: String, withDivider: Boolean = true) {
    Column {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
            Text(text = label, fontSize = FontSizes.text)
            Text(text = value, fontSize = FontSizes.largeText)
        }
        if(withDivider){
            Divider(modifier = Modifier.padding(vertical = Sizes.s8), color = Colors.subtitle)
        }
    }
}

File: ios/BoxRow.swift
//
//  BoxRow.swift
//  Expenses
//
//  Created by Aitor Cubeles Torres on 2/8/22.
//

import SwiftUI

struct BoxRow: View {
  let label: LocalizedStringKey
  let value: String
  let withDivider: Bool
  
  init(label: LocalizedStringKey, value: String) {
    self.label = label
    self.value = value
    self.withDivider = true
  }
  
  init(label: LocalizedStringKey, value: String, withDivider: Bool){
    self.label = label
    self.value = value
    self.withDivider = withDivider
  }
  
  var body: some View {
    VStack(spacing: 0) {
      HStack {
        Group {
          Text(label).font(.system(size: .FontSizes.text))
          Spacer()
          Text(value).font(.system(size: .FontSizes.largeText))
        }.foregroundColor(Color.Theme.text)
      }
      if(withDivider) {
        Divider().padding(.vertical, .Sizes.s8).foregroundColor(Color.Theme.subtitle)
      }
    }
  }
}
